Move jboss find commands to separate playbook and make non-default

noahl opened the pull request:
Two separate steps here:
  - Moving the JBoss find commands to their own playbook
  - Removing the scan facts from the default fact list

This has the result that Rho will not run a scan unless specifically
requested, and also the find commands are visibly separated from the
rest.

Closes #336 .

Commit:
Move jboss find commands to separate playbook and make non-default

Two separate steps here:
  - Moving the JBoss find commands to their own playbook
  - Removing the scan facts from the default fact list

This has the result that Rho will not run a scan unless specifically
requested, and also the find commands are visibly separated from the
rest.

chambridge commented on rho_playbook.yml:
```diff
@@ -22,6 +22,7 @@
     - virt
     - virt_what
     - jboss
+    - jboss-scan
```
I've never liked the role name of `jboss` when its more focused on *EAP*. If we are naming/renaming things could we update it to have an eap focus?

noahl replied:
Good point! I'll rename the fact to `jboss.eap.*` as we discussed. The folder can be something like `jboss-eap` or `jboss_eap`, unless you have a preference here.

chambridge commented on roles/jboss-scan/tasks/main.yml:
```diff
@@ -0,0 +1,14 @@
+# Scan linux systems for JBoss EAP or Wildfly Installations
+    - name: Gather jboss.jar-ver
+      raw: FOUND=""; for jar in `find {{scan_dirs}} -xdev -name 'jboss-modules.jar' 2>/dev/null | grep -v '\.installation/patches'`; do VERSION=$(java -jar ${jar} -version 2> /dev/null | grep version | sed 's/.*version\s//g'); inode=$(stat -c '%i' "${jar}"); fs=$(df  -T "${jar}" | grep "/dev" | sed 's/ .*//'); ctime=$(stat ${jar} | grep 'Change' | grep -oP '[1-2][0-9]{3}-[0-1][0-9]-[0-3][0-9]');  ...
+      register: jboss.jar-ver
```
wouldn't mind updating the fact names either to be EAP focus, but that could be for another PR.

noahl replied:
Will do.

chambridge reviewed (approved):
A few comments based on preference. But fine with this otherwise.

Commit:
Rename 'jboss' to 'jboss.eap'

Thanks to @chambrid for the review! Rename the plain 'jboss' facts to
'jboss.eap' to be clear about what they are detecting. Also rename the
folders to 'jboss_eap' to match, and update references to fact names
in the code.

chambridge commented on rho/utilities.py:
```diff
@@ -113,10 +113,10 @@
                       'subman.has_facts_file')
 
-JBOSS_FACTS_TUPLE = ('jboss.running-versions',)
+JBOSS_FACTS_TUPLE = ('jboss.eap.running-versions',)
```
Updates needed in `doc/source/command_syntax_usage.rst` with changed fact names.

Commit:
Update docs for jboss -> jboss.eap rename

Thanks to @chambrid for catching this.

Commit:
More doc updates for rename
